### ISSUE TITLE
Test cases where values with requires_grad = False flow through the network

### ISSUE THREAD

**Issue by jamesr66a** [pytorch/PiPPy#37]
For example, in our gradient accumulation (`accum_grads`) we might need to dynamically decide based on whether the lhs_grads or rhs_grads are `None`

**Comment by jamesr66a** [comment]
Surfaces that are affected by this:

null_coalesce_accumulate https://github.com/jamesr66a/PiPPy/blob/f07066f3f56f8f75b9a2032f26760dcd589653db/pippy/IR.py#L104
None checks in this fn: https://github.com/jamesr66a/PiPPy/blob/f07066f3f56f8f75b9a2032f26760dcd589653db/pippy/IR.py#L53
